blivet/fstab.py

```diff
@@ -0,0 +1,279 @@
+import os
+
+from libmount import Table, Fs, MNT_ITER_FORWARD
+from libmount import Error as LibmountException
+
+def parser_errcb(tb, fname, line):
+    print("{:s}:{:d}: parse error".format(fname, line))
+    return 1
+
+class FSTabManager():
+    """ Read, write and modify fstab file """
+
+    def __init__(self, src_file=None, dest_file=None):
+        self._table = Table()   # Space for parsed fstab contents
+
+        self.src_file = src_file
+
+        self.dest_file = dest_file
+        if dest_file is None:
+            dest_file = self.src_file
+
+        if self.src_file is not None:
+            self.read(self.src_file)
+
+    def _get_containing_device(self, path, devicetree):
+        """ Return the device that a path resides on. """
+        if not os.path.exists(path):
+            return None
+
+        st = os.stat(path)
+        major = os.major(st.st_dev)
+        minor = os.minor(st.st_dev)
+        link = "/sys/dev/block/%s:%s" % (major, minor)
+        if not os.path.exists(link):
+            return None
+
+        try:
+            device_name = os.path.basename(os.readlink(link))
+        except Exception:  # pylint: disable=broad-except
+            log_exception_info(fmt_str="failed to find device name for path %s", fmt_args=[path])
+            return None
+
+        if device_name.startswith("dm-"):
+            # have I told you lately that I love you, device-mapper?
+            # (code and comment copied from anaconda, kept for entertaining purposes)
+            device_name = blockdev.dm.name_from_node(device_name)
+
+        return devicetree.get_device_by_name(device_name)
```

Comment:
So the module will always require a populated devicetree to work? If that is true, we might be able to simplify some of the things above with something like `resolve_device`.

Reply:
It might not be that clear from anaconda perspective. I have added a new method called `find_device` that utilizes `resolve_device` and does not create the device if it does not exist.

Reply:
I have reworked the whole method, dropping some stuff that is probably not needed.